cli/pkg/kctrl/cmd/app/release/app_spec_builder.go

```diff
@@ -78,8 +78,8 @@ func (b *AppSpecBuilder) Build() (kcv1alpha1.AppSpec, error) {
 		if err != nil {
 			return kcv1alpha1.AppSpec{}, err
 		}
+		defer os.RemoveAll(LockOutputFolder)
```

Comment:
Let's also add a test for this (not deleting user supplied folders)?

Reply:
```suggestion
		defer os.RemoveAll(tmpImgpkgFolder)
```
I think it is more readable as we are creating `tmpImgpkgFolder` in `os.Mkdir(tmpImgpkgFolder, os.ModePerm)`

Reply:
Adding the test, meanwhile, I switched to using `LockOutputFolder` directly everywhere as it might or might not be temporary